server/clients/client4.go

```diff
@@ -0,0 +1,34 @@
+package clients
+
+import "github.com/mattermost/mattermost-server/v5/model"
+
+type Client4 interface {
+	SetToken(token string)
+	GetUserByUsername(username, etag string) (*model.User, *model.Response)
+	CreateBot(bot *model.Bot) (*model.Bot, *model.Response)
+	EnableBot(userID string) (*model.Bot, *model.Response)
+	GetUser(userID, etag string) (*model.User, *model.Response)
+	UpdateUserRoles(userID, newRoles string) (bool, *model.Response)
+	CreateUserAccessToken(userID, name string) (*model.UserAccessToken, *model.Response)
+	GetOAuthApp(clientID string) (*model.OAuthApp, *model.Response)
+	CreateOAuthApp(*model.OAuthApp) (*model.OAuthApp, *model.Response)
+}
+
+/*
+	asAdmin := model.NewAPIv4Client(conf.MattermostSiteURL)
+	asAdmin.SetToken(session.Token)
+*/
+
+type ClientService interface {
+	NewClient4(siteURL string) Client4
+}
```

Comment:
Is that level on direction really needed? Are additional clients expected to be added?